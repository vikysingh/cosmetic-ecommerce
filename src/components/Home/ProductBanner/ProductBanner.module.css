#PRODUCT_BANNER {
    width: 70vw;
    height: auto;
    margin: auto;
    /* background-color: pink; */
    padding: 50px 0;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
}

.PRODUCT_BANNER__card {
    width: 30%;
    height: 180px;
    text-align: center;
    opacity: 0.7;
    transition: 400ms all ease;
}

.PRODUCT_BANNER__card h3 {
    font-weight: lighter !important;
    margin-top: 10px;
}

.PRODUCT_BANNER__card img {
    width: 100%;
    height: 100%;
}

.PRODUCT_BANNER__card:hover {
    opacity: 1;
}

#PRODUCT_BANNER > .PRODUCT_BANNER__card:nth-child(2) > h3 {
    color: red;
}