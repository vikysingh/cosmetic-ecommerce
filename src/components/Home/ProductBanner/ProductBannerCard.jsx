import React from 'react'
import { Link } from "react-router-dom"

import styles from "./ProductBanner.module.css"
import { Typography } from "antd"
import routes from "../../../constants/routes.json"

import { connect } from "react-redux"

import { setProduct } from "../../../redux/actions/filters/filterActionGenerators"


const { Title } = Typography

function Productbannercard({ imgSrc, text, dispatch, productType}) {
    return (
        <Link to={routes.shop} className={styles.PRODUCT_BANNER__card}
        onClick={() => dispatch(setProduct(productType)) }
         >
            <img src={imgSrc} alt="" />
            <Title level={3}> { text } </Title>
        </Link>
    )
}

const mapStateToProps = state => state

export default connect(mapStateToProps)(Productbannercard)