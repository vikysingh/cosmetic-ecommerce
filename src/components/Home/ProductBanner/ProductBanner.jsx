import React from "react"

import { connect } from "react-redux"

import styles from "./ProductBanner.module.css"

import makeupEye from "../../../assets/images/makeup_eye.jpg"
import makeupLip from "../../../assets/images/makeup_lip2.jpg"
import makeupFace from "../../../assets/images/makeup_face.jpg"
import ProductStatics from "../../../constants/ProductStatics.json"

import ProductBannerCard from "./ProductBannerCard"
import { setProduct } from "../../../redux/actions/filters/filterActionGenerators"

function ProductBanner({ dispatch }) {
    return <div className={` ${styles.PRODUCT_BANNER} setSection flexCenterCenter`}  >
        <ProductBannerCard 
        title="COSMETIC EYES"
        imgSrc={makeupEye}
        setDispatch={() => dispatch(setProduct(ProductStatics.productType[3]))}
        text="EYES" />

        <ProductBannerCard 
        title="GLOW LIPS"
        imgSrc={makeupLip}
        setDispatch={() => dispatch(setProduct(ProductStatics.productType[7]))}
        text="AT BEST PRICES" />
        
        <ProductBannerCard 
        title="SHINING FACE"
        imgSrc={makeupFace}
        setDispatch={() => dispatch(setProduct(ProductStatics.productType[5]))}
        text="BEST QUALITY" />
        
    </div>
}

const mapStateToProps = state => state

export default React.memo(connect(mapStateToProps)(ProductBanner))