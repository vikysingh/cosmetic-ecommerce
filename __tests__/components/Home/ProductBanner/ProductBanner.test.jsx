import React from "react"
import { shallow } from "enzyme"

import ProductBanner from "../../../../components/Home/ProductBanner/ProductBanner"

test('Should render ProductBanner from Home', () => {
    const wrapper = shallow(<ProductBanner />)
    expect(wrapper).toMatchSnapshot()
})