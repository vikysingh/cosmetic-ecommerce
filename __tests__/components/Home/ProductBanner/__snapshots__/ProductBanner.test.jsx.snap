// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ProductBanner from Home 1`] = `
<div
  className="flexRowAroundCenter"
  id="PRODUCT_BANNER"
>
  <[object Object]
    imgSrc={null}
    productType="Eyeliner"
    text="EYES"
  />
  <[object Object]
    imgSrc={null}
    productType="Lipstick"
    text="LIPS"
  />
  <[object Object]
    imgSrc={null}
    productType="Foundation"
    text="FACE"
  />
</div>
`;
